spec/factories/self_disclosure_requests.rb

```diff
@@ -4,11 +4,17 @@
     status { "sent" }
   end
 
+  trait :sent do
+    after(:create, &:sent!)
+  end
+
   trait :received do
     status { "received" }
+    after(:create, &:received!)
```

Comment:
bit confused here - what does this do? Think it needs a comment ( are you trying to generate paper trail records?)